### PR TITLE
Dell s4100

### PR DESCRIPTION
**Pull request by lakesh-sharma**
Platform support for the DELL EMC S4100

### CODE REVIEW

File: machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# /etc/inittab for onie
```
Comment (cbrune) [inline machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab]: why is this file included?  I haven't seen that done before.

File: machine/dell/dellemc_s4100_c2338/machine.make [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+# Enable UEFI support
+UEFI_ENABLE = yes
+
+# Set Linux kernel version
+LINUX_VERSION		= 4.9
+LINUX_MINOR_VERSION	= 57
+
+# Override LSB_RELEASE_TAG
+override LSB_RELEASE_TAG = 3.33.1.1-6
```
Comment (cbrune) [inline machine/dell/dellemc_s4100_c2338/machine.make]: why LSB_RELEASE_TAG overridden ?  I haven't seen that done before.

File: machine/dell/dellemc_s4100_c2338/machine.make [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# http://www.iana.org/assignments/enterprise-numbers
+# DELL IANA number
+VENDOR_ID = 674
+
+I2CTOOLS_ENABLE = yes
+I2CTOOLS_SYSEEPROM = no
+
+CONSOLE_SPEED = 115200
+CONSOLE_DEV = 0
+CONSOLE_FLAG = 0
```
Comment (cbrune) [inline machine/dell/dellemc_s4100_c2338/machine.make]: CONSOLE_FLAG is ancient and does not do anything.  I would remove it.

File: machine/dell/dellemc_s4100_c2338/machine.make [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  $(error Unknown VENDOR_REV)
+endif
+
+# Vendor ID -- IANA Private Enterprise Number:
+# http://www.iana.org/assignments/enterprise-numbers
+# DELL IANA number
+VENDOR_ID = 674
+
+I2CTOOLS_ENABLE = yes
+I2CTOOLS_SYSEEPROM = no
```
Comment (cbrune) [inline machine/dell/dellemc_s4100_c2338/machine.make]: `I2CTOOLS_SYSEEPROM = no` is the now the default.  No need to set it.  I would remove this line.

**Comment by cbrune** [comment]
Some minor comments, but generally looks really good.

Regarding the pull request, it is weird to see commits  842372e, eb17760, cff8264.  I would only expect to see " Platform support for the DELL EMC S4100" f216ea3.

**Comment by cbrune** [comment]
In fact, having both 842372e and f216ea3 makes this unmergable.  I would squash all these commits down into a single "Platform support for DELL EMC S4100" commit.

**Comment by lakesh-sharma** [comment]
yes, i just want to commit machine/dell/dellemc_s4100_c2338, with commit
https://github.com/opencomputeproject/onie/pull/653/commits/f216ea38d888334c891139e6b71ec8d249c52515

Other 3 commits showing up could be due to the fetch/merge done in my old work-space ?
and should be ignored.

File: machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab (hunk 1 above)
Comment (lakesh-sharma) [reply machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab]: it is needed for micro-usb console to work , which as it uart-1

**Comment by lakesh-sharma** [comment]
Let me know if i need to re-submit the pull request by doing fresh fork. Not sure as why s4000 and other changes are showing up in pull request.
Pull request and desired changes needed are only for s4100

File: machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab (hunk 1 above)
Comment (cbrune) [reply machine/dell/dellemc_s4100_c2338/rootconf/sysroot-etc/inittab]: OK, I see.

At some point we may want to handle this in a more general way -- making a wholesale copy of inittab is a little dangerous.

Another approach would be to allow machines to *append* platform specific lines to the main inittab file at build time.  Somewhat like an include file mechanism.  Otherwise we run the risk of a machine missing out on some core ONIE updates to the inittab file.

**Comment by cbrune** [comment]
I can't really say what went wrong with your PR.  There definitely was some history in your fork from 2015.  I have a feeling some local work went into your fork's master branch back in 2015 and things went sideways.

I would close this PR and start over.

You can reset you fork's master branch to really be a clean copy of the upstream master branch, like this: (lifted from http://www-creators.com/en/archives/1687)

```sh
# 1) Check out your fork's local master branch.
$ git checkout master

# 2) Fetch commits from "upstream", and update all your remote tracking reference
$ git fetch upstream 

# 3) Force the local master to "upstream/master"
$ git reset --hard upstream/master
```

Now your fork's master branch should be clean.  From here precede as before, create a branch, do the work, make a PR.

**Comment by lakesh-sharma** [comment]
Closing this PR, as it seems to contain some changes, not needed to be committed.

**Comment by lakesh-sharma** [comment]
i tried the last steps, of resetting the local master, still wasn't able to commit.

Username for 'https://github.com': lakesh-sharma
Password for 'https://lakesh-sharma@github.com': 
To https://github.com/lakesh-sharma/onie.git
 ! [rejected]        dell_s4100 -> dell_s4100 (non-fast-forward)
error: failed to push some refs to 'https://github.com/lakesh-sharma/onie.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

**Comment by lakesh-sharma** [comment]
Also i tried -f option to forcing the commit.
git push -f origin dell_s4100_1,
with this the commit goes through, but the pull request of dell_s4100_1 branch still show the previous (not required) commits of 2015. It shows 4 commits, however we want just the last one to be attached with pull request. Moreover now dell_s4100_1 pull request message seems to stay, not sure as how to remove/delete it.


Username for 'https://github.com': lakesh-sharma
Password for 'https://lakesh-sharma@github.com': 
Counting objects: 22, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 2.16 KiB | 0 bytes/s, done.
Total 8 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To https://github.com/lakesh-sharma/onie.git
 * [new branch]      dell_s4100_1 -> dell_s4100_1
